python/tests/pyspark/lib/assertions.py

```diff
@@ -0,0 +1,35 @@
+import re
+from pandas.testing import assert_frame_equal
+
+
+def _print_dfs(actual, expected):
+    print('actual.show(100):')
+    actual.show(100, False)
+    print('expected.show(100):')
+    expected.show(100, False)
+
+
+def assert_df(actual, expected, sort=True, **kwargs):
+    """
+    For more elaborate DFs comparision, e.g. ignoring sorting or less precise.
+    Default precision for float comparison is 5 digits.
+    """
+    __tracebackhide__ = True
+
+    if sort:
+        actual = actual.orderBy(actual.columns)
+        expected = expected.orderBy(actual.columns)
+
+    try:
+        assert_frame_equal(actual.toPandas(), expected.toPandas(), **kwargs)
+    except AssertionError as e:
+        _print_dfs(actual, expected)
+        # assert_frame_equal doesn't print the column name, only the index
+        #   -> get the index from message with regex & resolve column name
+        matcher = re.search(r'iloc\[:, (\d+)\]', e.args[0])
```

Comment:
I'd avoid using regexes in code. They are hard to read and maintain: https://stackoverflow.com/questions/764247/why-are-regular-expressions-so-controversial

Reply:
The reason for having this is, as the comment says:

> assert_frame_equal doesn't print the column name, only the index

Should this use some other way for capturing the column name from the string than regex?